### PR TITLE
Dont use requireNothing

### PR DESCRIPTION
**Pull request by 45930**
Using `getAndRequireEquals` will create a precondition on the proof that the on-chain value matches the JS input.

Using `requireNothing` explicitly removes that precondition so any JS input will be valid.

### CODE REVIEW

File: contracts/src/blobstream/blobstream_contract.ts [hunk 1]
```diff
@@ -47,18 +47,16 @@ export class BlobstreamProcessor extends SmartContract {
 
     @method async update(admin: PrivateKey, blobstreamProof: BlobstreamProofType, path: BlobstreamMerkleWitness) {
         blobstreamProof.verify()
-        let leafIndex = this.currentLeafIndex.get();
-        this.currentLeafIndex.requireEquals(leafIndex);
+        let leafIndex = this.currentLeafIndex.getAndRequireEquals();
 
-        let commitmentsRoot = this.commitmentsRoot.get();
+        let commitmentsRoot = this.commitmentsRoot.getAndRequireEquals();
         this.commitmentsRoot.requireEquals(commitmentsRoot);
```
Comment (kobigurk) [inline contracts/src/blobstream/blobstream_contract.ts:53]: @45930 do we still need this line after you change?

Comment (45930) [reply contracts/src/blobstream/blobstream_contract.ts:53]: Ah, no.  `get()` and `requireEquals()`  can be replaced with `getAndRequireEquals()`

Comment (kobigurk) [reply contracts/src/blobstream/blobstream_contract.ts]: does it look correct now?

Comment (45930) [reply contracts/src/blobstream/blobstream_contract.ts]: Yeah, thanks!

**Pull request merged**